docs/src/main/asciidoc/deploying-to-google-cloud.adoc

```diff
@@ -0,0 +1,226 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Deploying to Google Cloud Platform (GCP)
+
+include::./attributes.adoc[]
+
+This guide covers:
+
+* Login to Google Cloud
+* Deploying a function to Google Cloud Functions
+* Deploying a JAR to Google App Engine Standard
+* Deploying a Docker image to Google App Engine Flexible Custom Runtimes
+* Deploying a Docker image to Google Cloud Run
+
+== Prerequisites
+
+For this guide you need:
+
+* Roughly 1 hour for all modalities
+* JDK 11
+* Apache Maven {maven-version}
+* https://cloud.google.com/[A Google Cloud Account].  Free accounts work.
+* https://cloud.google.com/sdk[Cloud SDK CLI Installed]
+
+This guide will take as input an application developed in the link:getting-started[Getting Started guide].
+
+Make sure you have the getting-started application at hand, or clone the Git repository: `git clone {quickstarts-clone-url}`,
+or download an {quickstarts-archive-url}[archive]. The solution is located in the `getting-started` directory.
+
+== Login to Google Cloud
+
+Login to Google Cloud is necessary for deploying the application and it can be done as follows:
+
+[source, subs=attributes+]
+----
+gcloud auth login
+----
+
+== Deploying to Google Cloud Functions
+
+Quarkus supports deploying your application to Google Cloud Functions via the following extensions:
+
+- link:gcp-functions[Google Cloud Functions]: Build functions using the Google Cloud Functions API.
+- link:gcp-functions-http[Google Cloud Functions HTTP binding]: Build functions using Quarkus HTTP APIs: RESTEasy (JAX-RS),
+Undertow (Servlet), Vert.x Web, or link:funqy-http[Funqy HTTP].
+- link:funqy-gcp-functions[Funky Google Cloud Functions]: Build functions using Funqy.
+
+Each extension support a specific kind of application development,
+follow the specific guides for more information on how to develop, package and deploy your application using them.
+
+== Deploying to Google App Engine Standard
+
+We will only cover the Java 11 runtime as the Java 8 runtime uses its own Servlet engine which is not compatible with Quarkus.
+
+First of all, make sure to have an App Engine environment initialized for your Google Cloud project, if not, initialize one via `gcloud app create --project=[YOUR_PROJECT_ID]`.
+
+Then, you will need to create a `src/main/appengine/app.yaml` file, let's keep it minimalistic with only the selected engine:
+
+[source, yaml]
+----
+runtime: java11
+----
+
+This will create a default service for your App Engine application.
+
+Then, you can choose to build the application by yourself or letting `gcloud` or the Google Cloud Maven build it for you.
```

Comment:
Google Cloud Maven **plugin**
maybe?

(Not sure)

Reply:
yes, done.